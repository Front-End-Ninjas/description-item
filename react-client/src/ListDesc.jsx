import React from 'react';
import PropTypes from 'prop-types';

const ListDesc = ({ item }) => (
  <div id="list_desc">
    <ul>
      <li id="list-item">{item}</li>
    </ul>
  </div>
);

export default ListDesc;

ListDesc.propTypes = {
  item: PropTypes.string.isRequired,
};
