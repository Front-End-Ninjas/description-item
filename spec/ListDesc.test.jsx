import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ListDesc from '../react-client/src/ListDesc';

describe('ListDesc Component', () => {
  const data = ['test'];

  it('should be defined', () => {
    expect(ListDesc).toBeDefined();
  });

  it('should render a <li> tag', () => {
    expect(shallow(<ListDesc item={data} />).find('#list-item').length).toEqual(1);
  });

  it('should renders correctly', () => {
    const tree = renderer.create(<ListDesc item={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
