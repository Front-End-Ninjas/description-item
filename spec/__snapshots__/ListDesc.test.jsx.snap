// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDesc Component should renders correctly 1`] = `
<div
  id="list_desc"
>
  <ul>
    <li
      id="list-item"
    >
      test
    </li>
  </ul>
</div>
`;
